### PR TITLE
A new package: py-openslide-python.

### PR DESCRIPTION
**Pull request by weijianwen**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-openslide-python/package.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    version('1.1.1', '8c207e48069887b63ea1c7bc9eb7dfc0')
+
+    install_time_test_callbacks = ['import_module_test']
+
+    import_modules = ['openslide']
+
+    depends_on('openslide@3.4.0:')
+    depends_on('python@2.6:2.8,3.3:')
+    depends_on('py-setuptools', type='build')
+    depends_on('py-pillow+jpeg+jpeg2000+tiff', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-openslide-python/package.py]: Are all of these variants required? Also, does it work with `py-pil` as well? We have a `pil` virtual dependency if so.

File: var/spack/repos/builtin/packages/py-openslide-python/package.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+class PyOpenslidePython(PythonPackage):
+    """OpenSlide Python is a Python interface to the OpenSlide library."""
+
+    homepage = "https://github.com/openslide/openslide-python"
+    url      = "https://github.com/openslide/openslide-python/archive/v1.1.1.tar.gz"
+
+    version('1.1.1', '8c207e48069887b63ea1c7bc9eb7dfc0')
+
+    install_time_test_callbacks = ['import_module_test']
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-openslide-python/package.py]: This is already in the PythonPackage base class, so you don't need to repeat it.

Comment (weijianwen) [reply var/spack/repos/builtin/packages/py-openslide-python/package.py]: Fixed.

File: var/spack/repos/builtin/packages/py-openslide-python/package.py (hunk 1 above)
Comment (weijianwen) [reply var/spack/repos/builtin/packages/py-openslide-python/package.py]: Those variants are required. `py-openslide-python` works with both `py-pil` and `py-pillow`. But variants `+jpeg+jpeg2000+tiff` are missing when specifying the virual dependency `pil`.

**Pull request merged**